src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -695,6 +696,42 @@ def load_pem_public_key(self, data):
             None,
         )
 
+    def load_der_private_key(self, data, password):
+        # Try to load it as a traditional or PKCS8 unencrypted key
+        ptr = self._ffi.new("unsigned char[]", data)
+        pp = self._ffi.new("unsigned char **", ptr)
+        key = self._lib.d2i_AutoPrivateKey(self._ffi.NULL, pp, len(data))
+        if key != self._ffi.NULL and password is not None:
+            raise TypeError(
+                "Password was given but private key is not encrypted."
+            )
+
+        if key != self._ffi.NULL:
+            return self._evp_pkey_to_private_key(key)
+        else:
+            # OpenSSL 0.9.8 can't load unencrypted PKCS8 keys using
+            # d2i_PKCS8PrivateKey_bio, but that case is covered by
+            # AutoPrivateKey above. So now we try to load it with the method
+            # that handles encrypted PKCS8 properly. But first, consume any
+            # errors the AutoPrivateKey may have added to the stack.
+            self._consume_errors()
+            return self._load_key(
+                self._lib.d2i_PKCS8PrivateKey_bio,
+                self._evp_pkey_to_private_key,
+                data,
+                password,
+            )
+
+    def load_der_public_key(self, data):
+        mem_bio = self._bytes_to_bio(data)
+        evp_pkey = self._lib.d2i_PUBKEY_bio(mem_bio.bio, self._ffi.NULL)
+        if evp_pkey == self._ffi.NULL:
+            self._consume_errors()
+            raise ValueError("Could not unserialize key data.")
+
+        evp_pkey = self._ffi.gc(evp_pkey, self._lib.EVP_PKEY_free)
+        return self._evp_pkey_to_public_key(evp_pkey)
```

Comment:
`_load_key` assumes it's going to load via bio with a function that takes an argument set that looks like (bio, null, password_callback, null). Since we don't have that here we can't use it.